File: apps/fluux/src-tauri/plugins/share-inbox/ios/Sources/ShareInboxPlugin.swift
import Foundation
import Tauri

class ShareInboxPlugin: Plugin {
    @objc public func inboxPath(_ invoke: Invoke) {
        guard let group = Bundle.main.object(forInfoDictionaryKey: "FluuxShareGroup") as? String,
              let root = FileManager.default.containerURL(forSecurityApplicationGroupIdentifier: group) else {
            invoke.reject("Shared container unavailable")
            return
        }
        invoke.resolve(["path": root.appendingPathComponent("ShareInbox").path])
    }
}
@_cdecl("init_plugin_share_inbox")
func initPlugin() -> Plugin { ShareInboxPlugin() }

File: apps/fluux/src-tauri/plugins/share-inbox/android/src/main/java/com/processone/shareinbox/ShareInboxPlugin.kt
package com.processone.shareinbox

import android.app.Activity
import app.tauri.annotation.Command
import app.tauri.annotation.TauriPlugin
import app.tauri.plugin.Plugin
import app.tauri.plugin.Invoke
import app.tauri.plugin.JSObject
import java.io.File

@TauriPlugin
class ShareInboxPlugin(private val host: Activity): Plugin(host) {
    @Command
    fun inboxPath(invoke: Invoke) {
        invoke.resolve(JSObject().put("path", File(host.filesDir, "ShareInbox").absolutePath))
    }
}
